code/sorting/quick_sort/quicksort.rs

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ 
+// This algorithm taken from https://github.com/servo/rust-quicksort
```

Comment:
This code is copied so it cannot be merged !
U need to do certain corrections to it ! :)